### PR TITLE
clone-deep npm package typings

### PR DESCRIPTION
**Pull request by tanfonto**
- [*] Use a meaningful title for the pull request. Include the name of the package modified.
- [*] Test the change in your own code. (Compile and run.)
- [*] Add or edit tests to reflect the change. (Run with `npm test`.)
- [*] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [*] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [*] Run `npm run lint package-name` (or `tsc` if no `tslint.json` is present).
- [*] The package does not already provide its own types, or cannot have its `.d.ts` files generated via `--declaration`
- [*] If this is for an NPM package, match the name. If not, do not conflict with the name of an NPM package.
- [*] Create it with `dts-gen --dt`, not by basing it on an existing project.
- [*] `tslint.json` should be present, and `tsconfig.json` should have `noImplicitAny`, `noImplicitThis`, `strictNullChecks`, and `strictFunctionTypes` set to `true`.

### CODE REVIEW

File: types/clone-deep/index.d.ts [hunk 1]
```diff
@@ -0,0 +1,14 @@
+// Type definitions for clone-deep 4.0
+// Project: https://github.com/jonschlinkert/clone-deep
+// Definitions by: Tanfonto <https://github.com/tanfonto>
+// Definitions: https://github.com/DefinitelyTyped/DefinitelyTyped
+// TypeScript Version: 2.7
+
+export interface InstanceClone<T> {
```
Comment (SimonSchick) [inline types/clone-deep/index.d.ts]: The interface name is longer than the function definition, I think you can just inline that.

**Review by SimonSchick — APPROVED** [review]

**Comment by octogonz** [comment]
@tanfonto these typings seem incorrect.

The **clone-deep** library exports a function:

```ts
function cloneDeep(val, instanceClone) {
   . . .
}
module.exports = cloneDeep;
```

...which should be imported using this TypeScript syntax:

```ts
import cloneDeep = require('clone-deep');
```

Whereas your typings describe a `default` export:

```ts
export default function cloneDeep<T>(
  val: T,
  instanceClone?: true | ((val: T) => T)
): T;
```
...which gives me this error:
``` error TS2349: Cannot invoke an expression whose type lacks a call signature. Type 'typeof import("@types/clone-deep/index")' has no compatible call signatures.
```

The correct typings should be:

```ts
declare function cloneDeep<T>(
  val: T,
  instanceClone?: true | ((val: T) => T)
): T;

export = cloneDeep;
```

**Comment by weswigham** [comment]
We'd take a PR to fix things like that. It's mostly my fault because I didn't catch it while reviewing.

**Comment by octogonz** [comment]
Just curious, how did the [unit tests](https://github.com/DefinitelyTyped/DefinitelyTyped/pull/33750/files#diff-ee6c5b4f66cafda2d39002b6a3d80ca2) pass?

**Comment by octogonz** [comment]
If I turn on `"esModuleInterop": true` in **tsconfig.json**, then these typings work correctly.  (But they are nonetheless incorrect.)

Oh yeah, I see that setting is enabled by the unit tests.  Perhaps DefinitelyTyped should use a lint rule to disable that.  It's a bad practice.

**Comment by weswigham** [comment]
DT unit tests aren't actually run, just typechecked

**Comment by octogonz** [comment]
@weswigham Here you go: https://github.com/DefinitelyTyped/DefinitelyTyped/pull/33990

**Pull request merged**